Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -481,11 +481,11 @@ public final class PackageBuilder {
                 }
 
                 // Make sure target is not referenced by absolute path
-                guard let relativeSubPath = try? RelativePath(validating: subpath) else {
+                if (try? RelativePath(validating: subpath)) == nil {
                     throw ModuleError.unsupportedTargetPath(subpath)
                 }
 
-                let path = packagePath.appending(relativeSubPath)
+                let path = AbsolutePath(subpath, relativeTo: packagePath)
```

Comment:
this change is fine, but its subtle and I can already see someone reverting this to use the nicer "appending" API at some future refactoring. in that sense it is related to our discussion on https://github.com/apple/swift-package-manager/pull/5446#pullrequestreview-962771235: we should really make `AbsolutePath::appending(RelativePath)` work well on Windows instead of deprecating it

Reply:
hmm, I _could_ see the implementation just being sugar for that potentially...

```
extension AbsolutePath {
  func appending(_ path: String) {
    AbsolutePath(path, relativeTo: self)
  }
}
```

Reply:
that is not a terrible idea IMO. @abertelrud ?

Reply:
Note that the type is `String` - we don't want to be able to use `RelativePath` unless we change the instances to `RelativePath(unsafeUncheckedPath:)` (since we cannot validate the path without the anchor).

Reply:
hmm, yea I think we want the `AbsolutePath::appending(RelativePath)` back, still unclear to me while we cannot use them since we are still creating an instance a couple of lines above

Reply:
The creation above only validates that the path does not start with `/` on Unix, and does not start with `[A-Z]:` or `\\` or `//`.  We could just inline that there I suppose.